apps/client/src/common/components/identify-overlay/IdentifyOverlay.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Badge } from '@chakra-ui/react';
+
+import { setClientRemote } from '../../hooks/useSocket';
+import { useClientStore } from '../../stores/clientStore';
+import { socketSendJson } from '../../utils/socket';
+
+import './Overlay.scss';
+
+export default function IdentifyOverlay() {
+  const { clients, id, name, setMyName } = useClientStore();
+  const { setIdentify, setRedirect, setRename } = setClientRemote;
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (id && clients[id] && clients[id].redirect !== '') {
+      const { redirect } = clients[id];
+      if (redirect !== window.location.pathname) {
+        setRedirect({ target: id, path: '' });
+        navigate(redirect);
+      } else {
+        setRedirect({ target: id, path: '' });
+      }
+    }
+  }, [clients, id, navigate, setRedirect]);
```

Comment:
I think we can simplify a bit
- avoid redundant calls
- exit early
It is also nice in `useEffects` to describe what they do

Something like (untested):
```suggestion
  // redirect if new target is set
  useEffect(() => {
    if (!id || !clients[id] || clients[id].redirect === '') {
      return;
    }

    const { redirect } = clients[id];
    if (redirect !== window.location.pathname) {
      setRedirect({ target: id, path: '' });
    }
  }, [clients, id, navigate, setRedirect]);
```